packages/runtime-core/src/componentProps.ts

```diff
@@ -370,7 +370,12 @@ function setFullProps(
       )
     }
   }
-
+  if (options) {
```

Comment:
This adds two additional object key enumerations each call which is not ideal - I managed to avoid them in 0255be2f